extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -48,14 +48,32 @@ export const signFreighterTransaction = createAsyncThunk<
 
 export const submitFreighterTransaction = createAsyncThunk<
   Horizon.TransactionResponse,
-  { signedXDR: string; networkDetails: NetworkDetails },
+  {
+    publicKey: string;
+    signedXDR: string;
+    networkDetails: NetworkDetails;
+    refreshBalances?: boolean;
```

Comment:
well I considered separating the actions into 2 actions, one where you can submit the tx standalone and one where you can submit the tx and refresh balances in one action. The reason is that I need a way to do both in one action lifecycle in order to avoid the "flash of screen" problem that happens if I first use the `submitFreighterTransaction` then the `getAccountBalances` since they have different lifecycles for each action(they each have their own pending/fulfilled/rejected). Typically I would do this with a layer that can compose actions like in redux-saga for example so maybe there is redux toolkit pattern for this that I haven't found yet?

Reply:
ah gotchya, I haven't used redux-saga before (looks cool) but, according to [RTK docs](https://redux.js.org/usage/side-effects-approaches) here looks like side effects are advised to be handled in middleware.

But in this case I'm not sure the side effect is needed? 🤔  Since `refreshBalances=true` is only being used in` TransactionDetails.index`, but the account balance isn't shown in the UI anywhere. Only when they go back to the /account view it's shown, but we're already dispatching getAccountBalances there.

wdyt? I may be missing something

Reply:
Oh yeah, so the subtlety here is that we need to know the updated balance of the asset from the tx in order to know wether we need to show the InfoBlock to suggest removing the trustline when a user's balance for an asset is empty due to the tx.

Reply:
ahh right I missed that. Ok that makes sense then thx for explaining